### PR TITLE
Split out federated PDU retrieval into a non-cached version

### PR DESCRIPTION
**Pull request by MadLittleMods**
Split out federated PDU retrieval into a non-cached version

Context: https://github.com/matrix-org/synapse/pull/11114/files#r741643968

This code is unused but thought it seemed like a small refactor with a little bit of value to split some of the logic out. And the caching/rate-limiting is handled in the `get_pdu` consumer.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] ~~Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)~~
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/federation/federation_client.py [hunk 1]
```diff
@@ -277,6 +277,60 @@ async def backfill(
...
+            SynapseError, NotRetryingDestination, FederationDeniedError
+        """
+
+        signed_pdu = None
+
+        transaction_data = await self.transport_layer.get_event(
+            destination, event_id, timeout=timeout
+        )
+
+        logger.info(
```
Comment (erikjohnston) [inline synapse/federation/federation_client.py:314]:
```suggestion
        logger.debug(
```

We will log the fact that we've requested the event elsewhere, so I don't think there is much need in upping the log level to `INFO`?

File: synapse/federation/federation_client.py [hunk 2]
```diff
@@ -277,6 +277,60 @@ async def backfill(
...
+            for p in transaction_data["pdus"]
+        ]
+
+        if pdu_list and pdu_list[0]:
+            pdu = pdu_list[0]
+
+            # Check signatures are correct.
+            signed_pdu = await self._check_sigs_and_hash(room_version, pdu)
+
+        return signed_pdu
```
Comment (erikjohnston) [inline synapse/federation/federation_client.py:332]:
```suggestion
            retuurn await self._check_sigs_and_hash(room_version, pdu)

        return None
```

And then we can remove the `signed_pdu = None` line at the top?
```suggestion
            signed_pdu = await self._check_sigs_and_hash(room_version, pdu)

        return signed_pdu
```

File: synapse/federation/federation_client.py (hunk 1 above)
Comment (MadLittleMods) [reply synapse/federation/federation_client.py:314]: Good catch, no need for this and just adjusted for debugging initially before

**Review by erikjohnston — APPROVED** [review]

**Comment by MadLittleMods** [comment]
Thanks for the review @erikjohnston 🦛

**Pull request merged**